temporal-remote-data-encoder/src/main/java/io/temporal/payload/codec/AbstractRemoteDataEncoderCodec.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.payload.codec;
+
+import com.google.protobuf.util.JsonFormat;
+import io.temporal.api.common.v1.Payload;
+import io.temporal.api.common.v1.Payloads;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+/**
+ * Performs encoding/decoding of the payloads via the Remote Data Encoder (RDE) available over http.
+ *
+ * <p>
+ *
+ * <h2>Remote Data Encoder Http Server specification</h2>
+ *
+ * <p>RDE Server must:
+ *
+ * <ul>
+ *   <li>Respond on URL paths ending with "/encode" and "/decode" for encoding and decoding
+ *       functionality respectively
```

Comment:
"ending with" implies there could be other stuff besides the base URL between it and the postfix.

Maybe instead `on the URL paths "$rdeUrl/encode" and "$rdeUrl/decode"`

Reply:
> "ending with" implies there could be other stuff besides the base URL between it and the postfix.
Yeah, this is actually true. There may be anything in between according to our spec. "/encode" and "/decode" don't have to have the same root.